src/audio/src/src_ipc4.c

```diff
@@ -99,17 +100,20 @@ int src_set_params(struct processing_module *mod, struct sof_sink *sink)
 
 	src_params.frame_fmt = valid_fmt;
 	ret = sink_set_params(sink, &src_params, true);
-
-	/* if module is to be run as DP, calculate module period
-	 * according to OBS size and data rate
-	 * as SRC uses period value to calculate its internal buffers,
-	 * it must be done here, right after setting sink parameters
-	 */
-	if (dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP)
+	if (dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP) {
+		/* if module is to be run as DP, calculate module period
+		 * according to OBS size and data rate
+		 * as SRC uses period value to calculate its internal buffers,
+		 * it must be done here, right after setting sink parameters
+		 */
 		dev->period = 1000000 * sink_get_min_free_space(sink) /
-		      (sink_get_frame_bytes(sink) * sink_get_rate(sink));
+			      (sink_get_frame_bytes(sink) * sink_get_rate(sink));
+		/* align down period to LL cycle time */
+		dev->period /= LL_TIMER_PERIOD_US;
+		dev->period *= LL_TIMER_PERIOD_US;
 
-	comp_info(dev, "SRC DP period calculated as: %u", dev->period);
+		comp_info(dev, "SRC DP period calculated as: %u", dev->period);
+	}
```

Comment:
Could this be a helper or otherwise shared code? Looks wrong to have DP constraints in multiple places in code. I.e. if the period needs to be aligned because of SRC specific reasons, it's good to have here, but if it's a limitation of DP infra, then it should be a common function used by all DP modules.

Reply:
unfortunately SRC module needs a calculated period for internal calculations at "set params". It happens a bit earlier than "prepare", when module adapter do a "general" period calculation

So, in case of current SRC implementation, it cannot be unified. 

in general, if a module wants to enforce its own period instead of a calculated one, it can do it just by setting dev->period to desired value. SRC does it - but for different reasons

Reply:
@singalsu fyi - some SRC updates may be needed for DP.

Reply:
Maybe calculating some things in SRC can be moved to prepare() if need. SRC needs a target period duration and it tries to match than with polyphase filter bank(s) constraints. In some cases with 44.1 family <--> 48 family rates those can't be met exactly.